src/cloudant/_common_util.py

```diff
@@ -213,6 +236,26 @@ def codify(code_or_str):
         return _Code(code_or_str)
     return code_or_str
 
+def execute_search_or_find_query(db_or_ddoc, url, data):
+    """
+    Provides a helper function to execute a query against the _find or _search
+    endpoint.
+    """
+    if isinstance(db_or_ddoc, Document):
+        # Set encoder variable if db_or_ddoc is a DesignDocument
+        encoder = db_or_ddoc.encoder
+    else:
+        # Set encoder variable if db_or_ddoc is a CloudantDatabase
+        encoder = db_or_ddoc.client.encoder
+    headers = {'Content-Type': 'application/json'}
+    resp = db_or_ddoc.r_session.post(
+        url,
+        headers=headers,
+        data=json.dumps(data, cls=encoder)
+    )
+    resp.raise_for_status()
+    return resp.json()
+
```

Comment:
I'm not a fan of this function.  Perhaps we can discuss this one when we talk later.